snapcraft/parts/setup_assets.py

```diff
@@ -33,23 +33,35 @@
 
 
 def setup_assets(
-    project: models.Project, *, assets_dir: Path, project_dir: Path, prime_dir: Path
+    project: models.Project,
+    *,
+    assets_dir: Path,
+    project_dir: Path,
+    prime_dir: Path,
+    meta_directory_handler: Optional[Callable[[Path, Path], None]] = None,
 ) -> None:
     """Copy assets to the appropriate locations in the snap filesystem.
 
     :param project: The snap project file.
     :param assets_dir: The directory containing snap project assets.
     :param project_dir: The project root directory.
     :param prime_dir: The directory containing the content to be snapped.
+    :param meta_directory_handler: handler for hooks built through the lifecycle,
```

Comment:
Pedantic: This is not intuitive (I cannot figure out what this function does based on the signature and docstring).

I think this should be more like `:param hook_handler: callback that overrides default behavior for handling hooks`

Reply:
it also handles gui, I see my comment is stale, will fix!